### PR TITLE
Rewrote README.md, design.md, and user.md for clarity and to better introduce Armada

### PR DESCRIPTION
**Pull request by severinson**
Currently, the purpose of Armada, and much of the design of Armada, is explained in the user guide (user.md). This PR moves the explanation of the purpose of Armada into the main readme (README.md) and consolidates the information about the design of Armada into design.md (including some information previously only available in the Armada helm docs). In addition, README.md, design.md, and user.md have been rewritten for clarity.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,39 +1,36 @@
-![Armada](./logo.svg)
+<img src="./logo.svg" width="200"/>
 
 [![CircleCI](https://circleci.com/gh/helm/helm.svg?style=shield)](https://circleci.com/gh/G-Research/armada)
 [![Go Report Card](https://goreportcard.com/badge/github.com/G-Research/armada)](https://goreportcard.com/report/github.com/G-Research/armada)
 
+Armada is a system for scheduling and running batch jobs (e.g., a compute job for training a machine learning model) over Kubernetes clusters. Armada is designed for high availability and to handle scheduling hundreds of jobs per second over thousands of nodes.
```
Comment (carlocamurri) [inline README.md:6]: Mention millions of jobs in queue

File: README.md [hunk 2]
```diff
@@ -1,39 +1,36 @@
...
-**Queue:** Represents a user or project, used to maintain fair share over time, with a priority factor
+For an overview of the architecture and design of Armada, and instructions for submitting jobs, see:
 
-**Job:** Unit of work to be run (described as a Kubernetes PodSpec)
+- [System overview](./docs/design.md)
+- [User guide](./user.md)
 
-**Job Set:** Group of related jobs. The API allows observing progress of a job set together
+For instructions of how to setup and develop Armada, see:
+- [Development quickstart](./docs/quickstart.md)
```
Comment (carlocamurri) [inline README.md:26]:
```suggestion
- [Quickstart](./docs/quickstart.md)
```

File: README.md [hunk 3]
```diff
@@ -1,39 +1,36 @@
...
-
-## Key concepts
+## Documentation
 
-**Queue:** Represents a user or project, used to maintain fair share over time, with a priority factor
+For an overview of the architecture and design of Armada, and instructions for submitting jobs, see:
 
-**Job:** Unit of work to be run (described as a Kubernetes PodSpec)
+- [System overview](./docs/design.md)
+- [User guide](./user.md)
```
Comment (carlocamurri) [inline README.md:23]:
```suggestion
- [User guide](./docs/user.md)
```

File: docs/design.md [hunk 4]
```diff
@@ -1,59 +1,227 @@
...
+If you just want to learn how to submit jobs to Armada, see:
 
-### Job Set
-All jobs are grouped into Job Sets with user-specified identifier. A Job Set represents a project or other higher level unit of work. Users can observe events in Job Sets through the API.
+- [User guide](./user.md)
+
+## Architecture
+
+Armada consists of two main components:
+- The Armada server, which is responsible for accepting jobs from users and deciding in what order, and on which Kubernetes cluster, jobs should run. Users submit jobs to the Armada server through the `armadactl` command-line utility or via a REST API.
```
Comment (carlocamurri) [inline docs/design.md:12]: Mention gRPC API

File: README.md (hunk 1 above)
Comment (severinson) [reply README.md:6]: Comment addressed by changing the first paragraph to:
`Armada is a system for scheduling and running batch jobs (e.g., a compute job for training a machine learning model) over Kubernetes clusters. Armada is designed for high availability and to handle scheduling hundreds of jobs per second (with potentially millions of jobs queued) over thousands of nodes.`

File: docs/design.md (hunk 4 above)
Comment (severinson) [reply docs/design.md]: Fixed

**Comment by severinson** [comment]
Thanks for comments @carlocamurri 
I've addressed them

**Review by d80tb7 — APPROVED** [review]

**Pull request merged**